File: solutions/python/pangram/1/pangram.py
def is_pangram(sentence):
    if sentence == "":
        return False

    characters = set()

    for letter in sentence:
        print('letter:',letter)
        if letter.isalpha() and letter not in characters:
            characters.add(letter.lower())

    if characters.__len__() == 26:
        return True
    else:
        return False

File: solutions/c/pangram/1/pangram.c
#include "pangram.h"
#include <stddef.h>

bool is_pangram(const char *sentence)
{
    if (sentence == NULL)
        return false;

    bool letters[26] = { false };
    const char *ptr = sentence;

    while (*ptr) {
        char ch = *ptr;
        if (ch >= 'A' && ch <= 'Z') {
            letters[ch - 'A'] = true;
        } else if (ch >= 'a' && ch <= 'z') {
            letters[ch - 'a'] = true;
        }
        ptr++;
    }

    for (int i = 0; i < 26; i++) {
        if (!letters[i])
            return false;
    }

    return true;
}
